planner/core/rule_partition_processor.go

```diff
@@ -60,16 +59,16 @@ func (s *partitionProcessor) rewriteDataSource(lp LogicalPlan) (LogicalPlan, err
 			// Union->(UnionScan->DataSource1), (UnionScan->DataSource2)
 			children := make([]LogicalPlan, 0, len(ua.Children()))
 			for _, child := range ua.Children() {
-				us := LogicalUnionScan{}.Init(ua.ctx)
+				us := LogicalUnionScan{conditions: p.conditions}.Init(ua.ctx)
```

Comment:
Is this line a bugfix? can we figure out a test case to cover it?

Reply:
Oh, another separate PR https://github.com/pingcap/tidb/pull/11187 is working on this.

Reply:
merged master branch